pkg/token/token.go

```diff
@@ -504,16 +477,103 @@ func NewVerifier(clusterID, partitionID, region string) Verifier {
 			},
 			Timeout: 10 * time.Second,
 		},
-		clusterID:         clusterID,
-		validSTShostnames: stsHostsForPartition(partitionID, region),
+		clusterID: clusterID,
+		partition: partitionID,
+		region:    region,
 	}
 }
 
-// verify a sts host, doc: http://docs.amazonaws.cn/en_us/general/latest/gr/rande.html#sts_region
+// Ensures that a given host is following the appropriate STS hostname format, e.g. sts.{region}.{suffix}.
+// Does not run any validation, so hostnames that contain invalid regions (e.g. sts.not-a-region.amazonaws.com) or
+// regions where STS/FIPS/Dualstack are not supported will be verified.
+// doc: http://docs.amazonaws.cn/en_us/general/latest/gr/rande.html#sts_region
 func (v tokenVerifier) verifyHost(host string) error {
-	if _, ok := v.validSTShostnames[host]; !ok {
```

Comment:
i think we should still store the value in cache such that this check can be quick and doesn't impact performance